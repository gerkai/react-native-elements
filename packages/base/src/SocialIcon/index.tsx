import { withTheme } from '../config';
import { SocialIcon, SocialIconProps } from './SocialIcon';

export { SocialIcon };
export type { SocialIconProps };
export default withTheme(SocialIcon, 'SocialIcon');
