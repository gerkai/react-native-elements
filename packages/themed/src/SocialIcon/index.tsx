import { withTheme } from '../config';
import {
  SocialIcon,
  SocialIconProps,
} from '@react-native-elements/base/dist/SocialIcon/SocialIcon';

export { SocialIcon };
export type { SocialIconProps };
export default withTheme<SocialIconProps>(SocialIcon, 'SocialIcon');
