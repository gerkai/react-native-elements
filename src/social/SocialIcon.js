import React, { PropTypes } from 'react'
import { View, StyleSheet, Platform, TouchableHighlight } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'
import Text from '../text/Text'
import fonts from '../config/fonts'
let styles

const colors = {
  facebook: '#3b5998',
  twitter: '#00aced',
  ['google-plus-official']: '#dd4b39',
  pinterest: '#cb2027',
  linkedin: '#007bb6',
  youtube: '#bb0000',
  vimeo: '#aad450',
  tumblr: '#32506d',
  instagram: '#517fa4',
  quora: '#a82400',
  foursquare: '#0072b1',
  wordpress: '#21759b',
  stumbleupon: '#EB4823',
  github: '#000000',
  ['github-alt']: '#000000',
  twitch: '#6441A5',
  medium: '#02b875',
  soundcloud: '#f50',
  gitlab: '#e14329',
  angellist: '#1c4082',
  codepen: '#000000'
}

const SocialIcon = ({component, type, button, onPress, iconStyle, style, iconColor, title, raised, light}) => {
  const Component = !onPress ? View : component || TouchableHighlight
  return (
    <Component
      underlayColor={light ? 'white' : colors[type]}
      onPress={onPress}
      style={[
        raised && styles.raised,
        styles.container,
        button ? styles.button : styles.icon,
        {backgroundColor: colors[type]},
        light && {backgroundColor: 'white'},
        light && !raised && {marginLeft: 2, marginRight: 2},
        style && style
      ]}>
      <View style={styles.wrapper}>
        <Icon
          style={[iconStyle && iconStyle]}
          color={light ? colors[type] : iconColor}
          name={type}
          size={24} />
        {
          button && title && (
            <Text style={[styles.title, light && {color: colors[type]}]}>{title}</Text>
          )
        }
      </View>
    </Component>
  )
}

SocialIcon.propTypes = {
  component: PropTypes.element,
  type: PropTypes.string,
  button: PropTypes.bool,
  onPress: PropTypes.func,
  iconStyle: PropTypes.any,
  style: PropTypes.any,
  iconColor: PropTypes.string,
  title: PropTypes.string,
  raised: PropTypes.bool
}

SocialIcon.defaultProps = {
  raised: true,
  iconColor: 'white'
}

styles = StyleSheet.create({
  container: {
    margin: 7,
    padding: 15,
    borderRadius: 30
  },
  raised: {
    ...Platform.select({
      ios: {
        shadowColor: 'rgba(0,0,0, .4)',
        shadowOffset: {height: 1, width: 1},
        shadowOpacity: 1,
        shadowRadius: 1
      },
      android: {
        elevation: 2
      }
    })
  },
  wrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  title: {
    color: 'white',
    marginLeft: 15,
    ...Platform.select({
      ios: {
        fontFamily: fonts.ios.black
      },
      android: {
        fontFamily: fonts.android.black
      }
    })
  },
  icon: {
    height: 52,
    width: 52
  }
})

export default SocialIcon
