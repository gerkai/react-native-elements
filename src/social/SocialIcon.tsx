import React from 'react';
import {
  View,
  StyleSheet,
  Platform,
  TouchableHighlight,
  ActivityIndicator,
  ViewStyle,
  StyleProp,
  TextStyle,
} from 'react-native';
import Icon from '../icons/Icon';
import Text from '../text/Text';
import fonts from '../config/fonts';
import { withTheme } from '../config';
import { RneFunctionComponent } from '../helpers';

const colors = {
  'github-alt': '#000000',
  'google-plus-official': '#dd4b39',
  google: '#dd4b39',
  'reddit-alien': '#fc461e',
  'stack-overflow': '#f27f33',
  angellist: '#1c4082',
  codepen: '#000000',
  envelope: '#000000',
  etsy: '#f2581e',
  facebook: '#4267B2',
  'facebook-messenger': '#0084ff',
  flickr: '#ff0084',
  foursquare: '#0072b1',
  github: '#000000',
  gitlab: '#e14329',
  instagram: '#517fa4',
  linkedin: '#007bb6',
  medium: '#02b875',
  pinterest: '#cb2027',
  quora: '#a82400',
  soundcloud: '#f50',
  steam: '#c6c3c1',
  stumbleupon: '#EB4823',
  tumblr: '#32506d',
  twitch: '#6441A5',
  twitter: '#00aced',
  vimeo: '#aad450',
  vk: '#5181b8',
  wechat: '#7bb32e',
  weibo: '#e6162d',
  whatsapp: '#075e54',
  wordpress: '#21759b',
  youtube: '#bb0000',
};

export type SocialMediaType =
  | 'facebook'
  | 'facebook-messenger'
  | 'whatsapp'
  | 'twitter'
  | 'google-plus-official'
  | 'google'
  | 'pinterest'
  | 'linkedin'
  | 'youtube'
  | 'vimeo'
  | 'tumblr'
  | 'instagram'
  | 'quora'
  | 'flickr'
  | 'foursquare'
  | 'wordpress'
  | 'stumbleupon'
  | 'github'
  | 'github-alt'
  | 'twitch'
  | 'medium'
  | 'soundcloud'
  | 'stack-overflow'
  | 'gitlab'
  | 'angellist'
  | 'codepen'
  | 'weibo'
  | 'vk';

export type SocialIconProps = {
  Component?: typeof React.Component;
  type?: SocialMediaType;
  button?: boolean;
  onPress?(): void;
  onLongPress?(): void;
  iconType?: string;
  iconStyle?: StyleProp<ViewStyle>;
  style?: StyleProp<ViewStyle>;
  iconColor?: string;
  underlayColor?: string;
  title?: string;
  raised?: boolean;
  disabled?: boolean;
  loading?: boolean;
  activityIndicatorStyle?: object | any[];
  small?: string;
  iconSize?: string | number;
  light?: boolean;
  fontWeight?: string;
  fontStyle?: StyleProp<TextStyle>;
  fontFamily?: string;
} & typeof defaultProps;

const defaultProps = {
  raised: true,
  iconType: 'font-awesome',
  iconColor: 'white',
  iconSize: 24,
  button: false,
};

const SocialIcon: RneFunctionComponent<SocialIconProps> = (props) => {
  const {
    activityIndicatorStyle,
    button,
    disabled,
    fontFamily,
    fontStyle,
    fontWeight,
    iconType,
    iconColor,
    iconSize,
    iconStyle,
    light,
    loading,
    onLongPress,
    onPress,
    Component = onPress || onLongPress ? TouchableHighlight : View,
    raised,
    small,
    style,
    title,
    type,
    underlayColor,
    ...attributes
  } = props;

  const shouldShowExpandedButton = button && title;

  return (
    <Component
      {...attributes}
      underlayColor={light ? 'white' : underlayColor || colors[type]}
      onLongPress={disabled ? null : onLongPress}
      onPress={disabled ? null : onPress}
      disabled={disabled}
      style={StyleSheet.flatten([
        raised && styles.raised,
        styles.container,
        button && styles.button,
        !button && raised && styles.icon,
        !button &&
          !light &&
          !raised && {
            width: iconSize * 2 + 4,
            height: iconSize * 2 + 4,
            borderRadius: iconSize * 2,
          },
        { backgroundColor: colors[type] },
        light && { backgroundColor: 'white' },
        style && style,
      ])}
    >
      <View style={styles.wrapper}>
        {(shouldShowExpandedButton || !loading) && (
          <Icon
            iconStyle={StyleSheet.flatten([iconStyle && iconStyle])}
            color={light ? colors[type] : iconColor}
            name={type}
            size={iconSize}
            type={iconType}
          />
        )}
        {shouldShowExpandedButton && (
          <Text
            style={
              StyleSheet.flatten([
                styles.title,
                light && { color: colors[type] },
                fontFamily && { fontFamily },
                fontWeight && { fontWeight },
                fontStyle && fontStyle,
              ]) as TextStyle
            }
          >
            {title}
          </Text>
        )}

        {loading && (
          <ActivityIndicator
            animating
            style={StyleSheet.flatten([
              styles.activityIndicatorStyle,
              activityIndicatorStyle,
            ])}
            color={light ? colors[type] : iconColor || 'white'}
            size={(small && 'small') || 'large'}
          />
        )}
      </View>
    </Component>
  );
};

SocialIcon.defaultProps = defaultProps;

const styles = StyleSheet.create({
  container: {
    margin: 7,
    borderRadius: 30,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    paddingTop: 14,
    paddingBottom: 14,
  },
  raised: {
    ...Platform.select({
      android: {
        elevation: 2,
      },
      default: {
        shadowColor: 'rgba(0,0,0, .4)',
        shadowOffset: { height: 1, width: 1 },
        shadowOpacity: 1,
        shadowRadius: 1,
      },
    }),
  },
  wrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    color: 'white',
    marginLeft: 15,
    ...Platform.select({
      android: {
        ...fonts.android.black,
      },
      default: {
        fontWeight: 'bold',
      },
    }),
  },
  icon: {
    height: 52,
    width: 52,
  },
  activityIndicatorStyle: {
    marginHorizontal: 10,
    height: 0,
  },
});

export { SocialIcon };
export default withTheme(SocialIcon, 'SocialIcon');
