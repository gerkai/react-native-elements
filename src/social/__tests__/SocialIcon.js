import React from 'react';
import {shallow} from 'enzyme';
import SocialIcon from '../SocialIcon';

describe('SocialIcon component', () => {
  it('should render without issues', () => {
    const component = shallow(<SocialIcon type="twitter"/>);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('should show loading indicator', () => {
    const component = shallow(<SocialIcon loading />);

    expect(component.find('ActivityIndicator').length).toBe(1);
  });

  it('should render light social icon', () => {
    const component = shallow(<SocialIcon
      light
      raised={false}
      type='medium'
    />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('should render social icon button', () => {
    const component = shallow(<SocialIcon
      title='Sign In With Facebook'
      button
      type='facebook'
    />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('should render social icon button', () => {
    const component = shallow(<SocialIcon
      title='Sign In With Facebook'
      button
      type='facebook'
    />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('should have onPress event', () => {
    const onPress = jest.fn();
    const component = shallow(<SocialIcon onPress={onPress} />);

    component.simulate('press');
    expect(onPress).toHaveBeenCalled();
  });
});
