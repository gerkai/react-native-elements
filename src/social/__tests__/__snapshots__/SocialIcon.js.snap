// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcon component should render without issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    disabled={false}
    onLongPress={[Function]}
    onPress={[Function]}
    style={
        Array [
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "alignItems": "center",
              "borderRadius": 30,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 7,
            },
            false,
            Object {
              "height": 52,
              "width": 52,
            },
            false,
            Object {
              "backgroundColor": "#00aced",
            },
            undefined,
            undefined,
          ]
    }
    underlayColor="#00aced"
>
    <View
        style={
            Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
        }
    >
        <Icon
            allowFontScaling={false}
            color="white"
            name="twitter"
            size={24}
            style={
                Array [
                    undefined,
                  ]
            }
        />
    </View>
</View>,
  "nodes": Array [
    <View
      disabled={false}
      onLongPress={[Function]}
      onPress={[Function]}
      style={
            Array [
                  Object {
                    "shadowColor": "rgba(0,0,0, .4)",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 1,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 1,
                  },
                  Object {
                    "alignItems": "center",
                    "borderRadius": 30,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "margin": 7,
                  },
                  false,
                  Object {
                    "height": 52,
                    "width": 52,
                  },
                  false,
                  Object {
                    "backgroundColor": "#00aced",
                  },
                  undefined,
                  undefined,
                ]
      }
      underlayColor="#00aced"
>
      <View
            style={
                  Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
            }
      >
            <Icon
                  allowFontScaling={false}
                  color="white"
                  name="twitter"
                  size={24}
                  style={
                        Array [
                              undefined,
                            ]
                  }
            />
      </View>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <SocialIcon
        button={false}
        iconColor="white"
        iconSize={24}
        raised={true}
        type="twitter"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "button": false,
          "iconColor": "white",
          "iconSize": 24,
          "raised": true,
          "type": "twitter",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          disabled={false}
          onLongPress={[Function]}
          onPress={[Function]}
          style={
                    Array [
                              Object {
                                "shadowColor": "rgba(0,0,0, .4)",
                                "shadowOffset": Object {
                                  "height": 1,
                                  "width": 1,
                                },
                                "shadowOpacity": 1,
                                "shadowRadius": 1,
                              },
                              Object {
                                "alignItems": "center",
                                "borderRadius": 30,
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "margin": 7,
                              },
                              false,
                              Object {
                                "height": 52,
                                "width": 52,
                              },
                              false,
                              Object {
                                "backgroundColor": "#00aced",
                              },
                              undefined,
                              undefined,
                            ]
          }
          underlayColor="#00aced"
>
          <View
                    style={
                              Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                      }
                    }
          >
                    <Icon
                              allowFontScaling={false}
                              color="white"
                              name="twitter"
                              size={24}
                              style={
                                        Array [
                                                  undefined,
                                                ]
                              }
                    />
          </View>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          disabled={false}
          onLongPress={[Function]}
          onPress={[Function]}
          style={
                    Array [
                              Object {
                                "shadowColor": "rgba(0,0,0, .4)",
                                "shadowOffset": Object {
                                  "height": 1,
                                  "width": 1,
                                },
                                "shadowOpacity": 1,
                                "shadowRadius": 1,
                              },
                              Object {
                                "alignItems": "center",
                                "borderRadius": 30,
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "margin": 7,
                              },
                              false,
                              Object {
                                "height": 52,
                                "width": 52,
                              },
                              false,
                              Object {
                                "backgroundColor": "#00aced",
                              },
                              undefined,
                              undefined,
                            ]
          }
          underlayColor="#00aced"
>
          <View
                    style={
                              Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                      }
                    }
          >
                    <Icon
                              allowFontScaling={false}
                              color="white"
                              name="twitter"
                              size={24}
                              style={
                                        Array [
                                                  undefined,
                                                ]
                              }
                    />
          </View>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SocialIcon
    button={false}
    iconColor="white"
    iconSize={24}
    raised={true}
    type="twitter"
/>,
}
`;
